pkg/v7/chartconfig/current.go

```diff
@@ -0,0 +1,76 @@
+package chartconfig
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1"
+	"github.com/giantswarm/errors/guest"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/controller/context/resourcecanceledcontext"
+	"github.com/giantswarm/tenantcluster"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/cluster-operator/pkg/label"
+)
+
+func (c *ChartConfig) GetCurrentState(ctx context.Context, clusterConfig ClusterConfig) ([]*v1alpha1.ChartConfig, error) {
+	c.logger.LogCtx(ctx, "level", "debug", "message", "looking for chartconfigs in the tenant cluster")
+
+	tenantG8sClient, err := c.newTenantG8sClient(ctx, clusterConfig)
+	if tenantcluster.IsTimeout(err) {
+		c.logger.LogCtx(ctx, "level", "debug", "message", "did not find the cluster-operator api cert in the Kubernetes API")
+
+		// We can't continue without the cert. We will retry during the next
+		// execution.
+		resourcecanceledcontext.SetCanceled(ctx)
+		c.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
+
+		return nil, nil
+
+	} else if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	listOptions := metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("%s=%s", label.ManagedBy, c.projectName),
+	}
```

Comment:
This is the only change in behaviour from the current resource. The operator only considers chartconfig CRs managed by the operator. This was a fix @tuommaki made for the PM with the hundreds of configmaps causing memory issues.

In this case I think its also useful for the "feature branch" deploy of charts. Currently cluster-operator will delete charts manually created in the giantswarm namespace. cc @puja108 @indyfree

Reply:
Nice! Had the label fix tuomas made already in mind for my issue. But now already resolved itself. I should go on more holidays and hope my issue magically resolve itself ;) :+1: